docs/source/algorithms/LinkedUBs-v1.rst

```diff
@@ -0,0 +1,177 @@
+
+    from mantid.simpleapi import * 
+    from math import sin, cos, pi 
+    import numpy as np 
+
+    # working directory (where UB is stored)
+    wDir='/home/xpb15563/Documents/Linked_UBs/Ruby/testing/'
+
+    # lattice parameters (for CalculateUMatrix)
+    a = 4.7750
+    b = 4.7750
+    c =  13.1154
+    alpha = 90
+    beta = 90
+    gamma = 120
+
+    # parameters for PredictPeaks 
+    MinDSpacing = 0.5
+    MaxDSpacing = 20
+    MinWavelength = 0.8
+    MaxWavelength = 9.3
+    ReflectionCondition='Primitive'
+
+    # parameters for LinkedUBs
+    QTolerance = 0.5
+    QDecrement = 0.95 
+    DTolerance = 0.02 
+    NumPeaks = 25
+    PeakIncrement = 10
+    Iterations = 10
+    DeleteWorkspace = False
+
+
+    # calculate phi axis at given omega
+    def CalculatePhiAxis(omega,chi):
+        """
+        calculates the vector associated with the phi axis ready for SetGoniometer
+        -omega axis is taken as (0,1,0)  in xyz right handed cartesian system
+        -the phi axis is first defined at omega = 270
+        -examples of useage:
+        -phi axis at (270.54) is given by CalculatePhiAxis(0,54)
+        -phi axis at (0,54) is given by CalculatePhiAxis(-90,54)
+        -phi axis at (90,54) is given by CalculatePhiAxis(-180,54)
+        """    
+        # omega axis
+        ux_omega = 0
+        uy_omega = 1
+        uz_omega = 0
+        u_omega = np.array([ux_omega,uy_omega,uz_omega])
+        # angle between phi axis and beam
+        chi = chi * pi/180
+        # phi axis at omega = 270
+        ux_phi = cos(chi)
+        uy_phi = sin(chi)
+        uz_phi = 0
+        u_phi = np.array([ux_phi, uy_phi, uz_phi])
+        # the omega angle at which to calculate the phi axis 
+        omega_axis_setting = omega * pi/180
+        # rotation around omega axis
+        c_omega = cos(omega_axis_setting)
+        s_omega = sin(omega_axis_setting) 
+        r1 = np.array([c_omega + ux_omega**2 * (1-c_omega), ux_omega * uy_omega * (1-c_omega) - uz_omega * s_omega, ux_omega * uz_omega * (1-c_omega) + uy_omega * s_omega])
+        r2 = np.array([uy_omega * ux_omega * (1-c_omega) + uz_omega * s_omega, c_omega + uy_omega**2 * (1-c_omega), uy_omega * uz_omega * (1-c_omega) - ux_omega * s_omega])
+        r3 = np.array([uz_omega * ux_omega * (1-c_omega) - uy_omega * s_omega, uz_omega * uy_omega * (1 - c_omega) + ux_omega * s_omega, c_omega + uz_omega**2 * (1-c_omega)])
+        r = np.vstack([r1, r2, r3])
+        # phi axis vector at given omega is the product of the rotation matrix and the initial phi axis vector at omega = 270
+        u_phi_rotated = np.dot(r, u_phi)
+        u_phi_rotated = np.round(u_phi_rotated,5)
+        return u_phi_rotated 
+
+
+    # calculate phi axis at omega = 270 
+    u_phi_x, u_phi_y, u_phi_z = CalculatePhiAxis(0,54)
+
+    # load and process 41598
+    LoadRaw(Filename='/archive/NDXWISH/Instrument/data/cycle_18_1/WISH00041598.raw', OutputWorkspace='WISH00041598')
```

Comment:
I'm afraid we can't use any facility archive data in our tests as they tests have to be able to run anywhere, including outside of a facility where the archive access is not available. The WISH datasets are also quite large and we tend to prefer using smaller data sets for these tests if possible as they data needs to be shipped around.

Would it be possible to use basically the test in the system test that you have added with the SXD data? If we wanted to keep the WISH example as well then we could use a `.. code-block:: python` instead of a `.. testcode::` as in [this example](https://raw.githubusercontent.com/mantidproject/mantid/d87b6fe4ac41132f0b44e68cd85d80a07f12b4f2/docs/source/algorithms/LoadEventAndCompress-v1.rst) and then this will display the code on the docs page but not run the test. It would be acceptable to add the WISH test as a system test thought as they are allowed to run for longer and use larger data sets.

Reply:
Thanks Martyn, I have changed `.. testcode::` to `.. code-block:: python`. 

Regarding the failed RHEL7+System Tests, there seems to be an issue with using conventional `a, b, c, alpha, beta, gamma` as lattice parameters. Should I change their names in the algorithm or are they ok as is?

Reply:
Thanks. Could you remove the "/archive/..." bits from the path to and just leave the file name? The docs are publicly visible and it would be nice not do have too much of the ISIS data archive structure visible in the public documents.

Regarding the parameters, usually I would say change them but I think in this case we can make an exception as `SetUB` uses the standard convention of lower-case names too. Please could you add an exception for this algorithm in `https://github.com/mantidproject/mantid/blob/master/Testing/SystemTests/tests/analysis/CodeConventions.py#L49`? [System Tests](http://developer.mantidproject.org/SystemTests.html#selecting-tests-to-run) on the developer pages describes how to run those tests locally so you can check that you have fixed it before pushing.

Reply:
Thanks, I will do. 

Regarding the .rst file, is it necessary to have the `.. testcode::` in there? Is this an additional check that gets carried out similar to the system test?

Reply:
Thanks.

The `.. testcode::` blocks are only required when we want to run a test on that block of code within the documentation in addition to rendering it on the page to look like code and not normal text.